File: leetcode/subtract_product_and_sum_of_digits.js
// https://leetcode.com/problems/subtract-the-product-and-sum-of-digits-of-an-integer/

var subtractProductAndSum = function(n) {
    const digits = Array.from(String(n), Number)
    const sum = digits.reduce((a,b) => a+b)
    const product = digits.reduce((a,b) => a*b)
    return product-sum
};

File: leetcode/subtract_product_and_sum_of_digits.rb
# https://stackoverflow.com/questions/47953598/rubys-digits-method-performance

def subtract_product_and_sum(n)
   digits = n.to_s.chars.map(&:to_i)
   sum = digits.sum
   product = digits.reduce(&:*)
   product-sum
end

#one-liner // but digits is slower
# def subtract_product_and_sum(n)
#    n.digits.reduce(&:*) - n.digits.sum
# end
